### PR TITLE
Reflections

### PR DESCRIPTION
**Pull request by yalania**
- Added SSR (screen space reflections) to have some reflections, they are not mirror like reflections and so they are not perfect, there are some values we can tweak
- Added skybox reflections

Note: Merge after fog branch

### CODE REVIEW

File: Game/Resources/Shaders/Lit/blinn_phong.fs [hunk 1]
```diff
@@ -161,6 +165,10 @@ in vec4 position_full_depth_space;
 
 layout (location = 0) out vec4 FragColor;
 layout (location = 1) out vec4 BrightColor;
+layout (location = 2) out vec4 normalBuffer;
+layout (location = 3) out vec4 positionBuffer;
+layout (location = 4) out vec4 ssrValuesBuffer;
```
Comment (raulgonzalezupc) [inline Game/Resources/Shaders/Lit/blinn_phong.fs:173]: Can you put this variables on CamelCase like the others to be consistent please? ^^

**Review by mariofv — COMMENTED** [review]
Scene window doesn't render

File: Game/Resources/Shaders/Lit/blinn_phong.fs [hunk 2]
```diff
@@ -206,9 +214,14 @@ void main()
  	float lit_fragment = ShadowCalculation();
 
 	result += diffuse_color.rgb * ambient * occlusion_color.rgb; //Ambient light
-
+#if ENABLE_LIQUID_PROPERTIES
+    vec3 I = normalize(position - view_pos);
+    vec3 R = reflect(I, fragment_normal);
+	result +=texture(skybox_texture, R).rgb* 0.5;
```
Comment (raulgonzalezupc) [inline Game/Resources/Shaders/Lit/blinn_phong.fs]: you are adding the skybox texture allways without having in consideration if it is ocluded by another object.

Comment (raulgonzalezupc) [reply Game/Resources/Shaders/Lit/blinn_phong.fs]: there should be and option to not have reflections of the skybox at all in my opinion, or at least be able to tweak the value of 0.5

**Review by raulgonzalezupc — CHANGES_REQUESTED** [review]
It's kind of working as you said, there are a lot to improve, but if you make the changes requested I will aproved, I already tested but it feels kind of weird, and for our game I don't think the water will be super reflective since we have a cartoonish style. But pretty good job as allways ^^

**Comment by raulgonzalezupc** [comment]
You have to change the code style of the layouts of the blinn still but I aprove it

**Review by raulgonzalezupc — APPROVED** [review]

**Review by mariofv — APPROVED** [review]
Water looks like water now, 10/10

**Pull request merged**